util/report_utils/build_scale_csv_summary.rb

```diff
@@ -24,8 +20,134 @@
+
+  # create a summary csv file to contain the failing iteration from each result
+  @summary_csv_path_fail = "#{parent_dir}/#{parent_name}.summary_fail.csv"
+  create_summary_csv(@summary_csv_path_fail)
+
+  # start with the template
+  report = File.read(@template_path)
+
+  # process each result dir
+  process_result_dirs(parent_dir)
+
+  # calculate variance for success
+  calculate_summary_csv(@summary_csv_path_success)
+
+  # calculate variance for failure
+  calculate_summary_csv(@summary_csv_path_fail)
+
+  # csv2html
+  csv2html(@summary_csv_path_success)
+  csv2html(@summary_csv_path_fail)
+
+  # result name
+  report = report.gsub("$RESULT_NAME", parent_name)
+
+  # add summary success table
+  summary_success_table = extract_table_from_csv2html_output("#{@summary_csv_path_success}.html")
+  report = report.gsub("$RESULT_SUMMARY_SUCCESS_TABLE", summary_success_table)
+
+  # add summary fail table
+  summary_fail_table = extract_table_from_csv2html_output("#{@summary_csv_path_fail}.html")
+  report = report.gsub("$RESULT_SUMMARY_FAIL_TABLE", summary_fail_table)
+
+  # add extract tables
+  report = report.gsub("$RESULT_EXTRACT_TABLES", @tables)
+
+  # write report
+  puts "writing report to #{output_path}"
+  puts
+
+  File.write(output_path, report)
+end
+
+# Create the summary CSV file
+#
+# @author Bill Claytor
+#
+# @param [String] summary_csv_path The summary CSV file path
+#
+# @return [void]
+#
+# @example
+#   create_summary_csv(summary_csv_path)
+#
+def create_summary_csv(summary_csv_path)
+  puts "creating summary csv: #{summary_csv_path}"
+
+  CSV.open(summary_csv_path, "wb") do |csv|
+    csv << SUMMARY_CSV_HEADINGS
+  end
+end
+
+# Extract data from each result dir in the parent dir to build the report
+#
+# @author Bill Claytor
+#
+# @param [String] parent_dir The parent directory containing the scale test result directories
+#
+# @return [void]
+#
+# @example
+#    process_result_dirs(parent_dir)
+#
+def process_result_dirs(parent_dir)
+  # get all the result dirs in the parent dir
+  results_dirs = Dir.glob("#{parent_dir}/*/").sort
+  puts "processing #{results_dirs.length} results directories..."
+
+  # process each result dir
+  results_dirs.each do |dir|
+    dirname = File.basename(dir)
+    puts "processing results dir: #{dirname}"
+
+    csv_path = "#{dir}#{dirname}.csv"
+    puts "csv_path: #{csv_path}"
+
+    # create a summary csv with the headings and last 2 rows
+    heading = "<h1>#{dirname}</h1>"
+    table = process_results_csv(csv_path)
+    @tables += heading + table unless table.nil?
+  end
+end
 
-def process_results_csv(csv_path, summary_csv_path)
+# Extract data from the specified scale test results CSV file
+#
+# @author Bill Claytor
+#
+# @param [String] csv_path The scale test results CSV file path
+#
+# @return [String] The CSV data converted to an HTML table
+#
+# @example
+#    table = process_results_csv(csv_path)
+#
+def process_results_csv(csv_path)
```

Comment:
nitpick: You have one `process_*` method that has a return value (process_result_csv) and one that has no return value (process_result_dirs).  This UX is inconsistent.  Consider altering the naming convention for these methods.

Reply:
Renamed `process_result_csv` to `extract_table_from_results_csv`.